src/app/firebase_app.ts

```diff
@@ -516,6 +526,12 @@ export function createFirebaseNamespace(): FirebaseNamespace {
     // Monkey-patch the serviceNamespace onto the firebase namespace
     (namespace as any)[name] = serviceNamespace;
 
+    // Patch the FirebaseAppImpl prototype
+    FirebaseAppImpl.prototype[name] = function(...args) {
+      const serviceFxn = this._getService.bind(this, name);
+      return serviceFxn.apply(this, allowMultipleInstances ? args : []);
+    }
```

Comment:
Great catch on this. I have refactored to ensure that we are passing through the `instanceIdentifier` I must have missed it in adding code back.

In addition I have added an additional assertion to two of the tests below to ensure that we don't break this going forward.